hpy/tools/autogen/public_api.h

```diff
@@ -402,6 +402,54 @@ HPy HPy_Type(HPyContext *ctx, HPy obj);
 HPy_ID(166)
 int HPy_TypeCheck(HPyContext *ctx, HPy obj, HPy type);
 
+/**
+ * Return the type's name.
+ *
+ * Equivalent to getting the type's ``__name__`` attribute. If you want to
+ * retrieve the type's name as Python unicode object, then just use
+ * ``HPy_GetAttr_s(ctx, type, "__name__")``.
+ *
+ * :param ctx:
+ *     The execution context.
+ * :param type:
+ *     A Python type object. This argument must not be ``HPy_NULL`` and must be
+ *     a type (i.e. it must inherit from Python ``type``). If this is not the
+ *     case, the behavior is undefined (verification of the argument is only
+ *     done in debug mode).
+ *
+ * :returns:
+ *     The name of the type as C string (UTF-8 encoded) or ``NULL`` in case of
+ *     an error. The returned pointer is read-only and guaranteed to be valid as
+ *     long as the handle ``type`` is valid.
+ */
+HPy_ID(253)
+const char *HPyType_GetName(HPyContext *ctx, HPy type);
```

Comment:
Don't we have a utf8_t typedef to indicate the encoding?

Reply:
No, we didn't use that but named the arguments like `utf8_something`. That's, ofc, not possible for return types. The docs give the necessary hint. I hope that is enough.

Reply:
OK, thanks